tests/test_requests.py

```diff
@@ -315,6 +315,26 @@ def test_headers_on_session_with_None_are_not_sent(self, httpbin):
         prep = ses.prepare_request(req)
         assert 'Accept-Encoding' not in prep.headers
 
+    def test_headers_preserve_order(self, httpbin):
+        """Preserve order when headers provided as OrderedDict."""
+        ses = requests.Session()
+        ses.headers = OrderedDict()
+        ses.headers['Accept-Encoding'] = 'identity'
+        ses.headers['First'] = '1'
+        ses.headers['Second'] = '2'
+        headers = OrderedDict([('Third', '3'), ('Fourth', '4')])
+        headers['Fifth'] = '5'
+        headers['Second'] = '222'
+        req = requests.Request('GET', httpbin('get'), headers = headers)
+        prep = ses.prepare_request(req)
+        items = prep.headers.items()
```

Comment:
You either need to set your expected headers as a list and then zip the items of that list together with the output from `items()` here or you need to call `list(prep.headers.items())` because an `ItemsView` is not indexable.